subworkflows/local/filter_variants/main.nf

```diff
@@ -0,0 +1,37 @@
+include { ENSEMBLVEP_FILTERVEP          } from '../../../modules/nf-core/ensemblvep/filtervep/main'
+include { BCFTOOLS_VIEW                 } from '../../../modules/nf-core/bcftools/view/main'
+
+workflow FILTER_VARIANTS {
+
+    take:
+    ch_vcf      // channel  [optional] [ val(meta), path(vcf) ]
+    ch_hgnc_ids // channel: [optional] [ val(meta), path(txt) ]
+    filter_hgnc //    bool: should filter_vep be run to filter on hgnc ids
+
+    main:
+    ch_versions = Channel.empty()
+
+    if ( filter_hgnc ) {
+
+
+        ENSEMBLVEP_FILTERVEP (
+            ch_vcf,
+            ch_hgnc_ids.map { meta, file -> file }
+        )
+        ch_versions = ch_versions.mix(ENSEMBLVEP_FILTERVEP.out.versions)
+    }
+
+    BCFTOOLS_VIEW (
+        filter_hgnc ? ENSEMBLVEP_FILTERVEP.out.output.map { meta, vcf -> [ meta, vcf, [] ] } : ch_vcf.map { meta, vcf -> [ meta, vcf, [] ] },
```

Comment:
I think that it would be clearer to set the output in the `filter_hgnc` block to overwrite `ch_vcf` so that the input for bcftools_view always is `ch_vcf.map { meta, vcf -> [ meta, vcf, [] ] }`